app/src/main/java/com/github/se/project/ui/overview/Overview.kt

```diff
@@ -531,3 +597,69 @@ private data class SectionInfo(
     val icon: ImageVector,
     val tutorEmpty: Boolean = false
 )
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun CancellationAlerts(
+    profile: Profile,
+    lessons: List<Lesson>,
+    listProfilesViewModel: ListProfilesViewModel,
+    lessonViewModel: LessonViewModel
+) {
+  lessons.forEach { lesson ->
+    val isTutor = profile.role == Role.TUTOR
+
+    val tutor =
+        if (isTutor) profile
+        else
+            listProfilesViewModel.getProfileById(lesson.tutorUid.first())
+                ?: return Text("Tutor not found")
+
+    val student =
+        if (isTutor)
+            listProfilesViewModel.getProfileById(lesson.studentUid)
+                ?: return Text("Student not found")
+        else profile
```

Comment:
May be you could have create a component because the overview is already a big file and it would improve readability.